src/common/src/config.rs

```diff
@@ -921,10 +921,23 @@ pub struct S3ObjectStoreConfig {
     pub object_store_req_retry_max_delay_ms: u64,
     #[serde(default = "default::object_store_config::s3::object_store_req_retry_max_attempts")]
     pub object_store_req_retry_max_attempts: usize,
+    #[serde(default)]
+    pub developer: S3ObjectStoreDeveloperConfig,
+}
+
+/// The subsections `[storage.object_store.s3.developer]`.
+#[derive(Clone, Debug, Serialize, Deserialize, DefaultFromSerde)]
+pub struct S3ObjectStoreDeveloperConfig {
     /// Whether to retry s3 sdk error from which no error metadata is provided.
-    #[serde(default = "default::object_store_config::s3::retry_unknown_service_error")]
+    #[serde(
+        default = "default::object_store_config::s3::developer::object_store_retry_unknown_service_error"
+    )]
     pub retry_unknown_service_error: bool,
```

Comment:
Can we add an alias for this config for backward compatibility? Some PoC user already uses the original config without the `developer` scope.

Reply:
![Screenshot 2024-01-23 at 6 38 54 PM](https://github.com/risingwavelabs/risingwave/assets/47273164/7e5b0ebe-5019-47cb-b921-b7fe8269d1dc)

Done